lib/tests/CMakeLists.txt

```diff
@@ -18,16 +18,22 @@ set(TEST_CASES
     program_options.cc
 )
 
+link_directories(${Boost_LIBRARY_DIRS})
+
 add_executable(lib${PROJECT_NAME}_test $<TARGET_OBJECTS:libprojectsrc> ${TEST_CASES} main.cc)
 
 if (${CMAKE_SYSTEM_NAME} MATCHES "Linux" OR ${CMAKE_SYSTEM_NAME} MATCHES "SunOS")
     target_link_libraries(lib${PROJECT_NAME}_test rt)
 endif()
 
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND BOOST_STATIC)
+    target_link_libraries(lib${PROJECT_NAME}_test iconv)
+endif()
+
 target_link_libraries(lib${PROJECT_NAME}_test
     ${Boost_LIBRARIES}
-    ${LEATHERMAN_LIBRARIES}
     ${CMAKE_THREAD_LIBS_INIT}
+    ${LEATHERMAN_LIBRARIES}
```

Comment:
Link order matters. I'd put iconv after Boost_LIBRARIES (probably doesn't matter where it is related to LEATHERMAN_LIBRARIES, but after them as well makes sense to me).